workflows/summary_stats.snake

```diff
@@ -0,0 +1,291 @@
+
+# Select DFE model from catalog  
+dfe_list = config["dfe_list"]
+annotation_list = config["annotation_list"]
+mask_file = config["mask_file"]
+
+
+
+nchunks=100
+chunks = np.arange(nchunks)
+# ###############################################################################
+# GENERAL RULES & GLOBALS
+# ###############################################################################
+
+
+rule all:
+    input:
+        expand(
+            [output_dir + "/summaries/{{demog}}/{dfes}/{annots}/{{seeds}}/sim_{{chrms}}.{{exts}}".format(
+                dfes=DFE, annots=ANNOT) for (DFE, ANNOT) in zip(dfe_list, annotation_list)],
+            demog=demo_model_ids,
+            seeds=seed_array,
+            chrms=chrm_list,
+            exts=["sampleToPopFile", 
+                "ancestralAllelesFile", 
+                "vcf",
+                ],
+        ),
+       # expand(
+       #     [output_dir + "/summaries/{demog}/{dfes}/{annots}/{{seeds}}/sim_{{chrms}}.{popid}.{{chunk}}.{{exts}}".format(
+       #         dfes=DFE, annots=ANNOT, popid=POPID, demog=DEMOG)
+       #             for (POPID, DEMOG, DFE, ANNOT) in [item 
+       #                 for xx in [[(i, z[0],z[1][0],z[1][1]) 
+       #                 for i in demo_pop_ids[z[0]]] 
+       #                 for z in [ x 
+       #                     for x in itertools.product(demo_model_ids, zip(dfe_list, annotation_list))]] 
+       #                     for item in xx]],
+       #     seeds=seed_array,
+       #     chrms=chrm_list,
+       #     chunk=chunks,
+       #     exts=["diploshic.fvec", "diploshic.stats"],
+       # ),
+        expand(
+            [output_dir + "/summaries/{demog}/{dfes}/{annots}/{{seeds}}/sim_{{chrms}}.{popid}.{{exts}}".format(
+                dfes=DFE, annots=ANNOT, popid=POPID, demog=DEMOG) for (POPID, DEMOG, DFE, ANNOT) in [item for xx in [[(i, z[0],z[1][0],z[1][1]) for i in demo_pop_ids[z[0]]] for z in [ x for x in itertools.product(demo_model_ids, zip(dfe_list, annotation_list))]] for item in xx]],
+            seeds=seed_array,
+            chrms=chrm_list,
+            chunk=chunks,
+            exts=["pi.png",
+                 "diploshic.stats",
+                 ],
+        ),
+        expand(
+            [output_dir + "/summaries/{demog}/{dfes}/{annots}/sim_{{chrms}}.allseeds.{popid}.pi.png".format(
+                dfes=DFE, annots=ANNOT, popid=POPID, demog=DEMOG) for (POPID, DEMOG, DFE, ANNOT) in [item for xx in [[(i, z[0],z[1][0],z[1][1]) for i in demo_pop_ids[z[0]]] for z in [ x for x in itertools.product(demo_model_ids, zip(dfe_list, annotation_list))]] for item in xx]],
+            chrms=chrm_list,
+        ),
+
+
+
+ 
+
+rule make_vcf:
+    input:
+        output_dir + "/simulated_data/{demog}/{dfes}/{annots}/{seeds}/sim_{chrms}.trees"
+
+    output:
+        output_dir + "/summaries/{demog}/{dfes}/{annots}/{seeds}/sim_{chrms}.vcf"
+    run:
+        ts = tskit.load(input[0])
+        ts.write_vcf(open(output[0], "w"))
```

Comment:
I think this would be cleaner, since then I am sure the file is closed, as 
```
with open(output[0], 'w') as vcf:
    ts.write_vcf(vcf, contig_id=wildcards.chrms)
```
If you add the contig_id then you can cat the diploshic output files for each population in a seed. Then when plotting set the chromosome as a hue or in relplot as rows.

Reply:
if you add this change you also have to add wildcards.chrms to the ancestralAllelesFile so it assoc w/ the correct ancestral file. However, then there is another error. 
```
Traceback (most recent call last):                                                                                                                                                                                 
  File "/home/ssmall/mambaforge/envs/analysis2/bin/makeFeatureVecsForChrArmFromVcf_ogSHIC.py", line 129, in <module>                                                                                               
    genos = allel.GenotypeArray(genos.subset(sel0=snpIndicesToKeep))                                                                                                                                               
  File "/home/ssmall/mambaforge/envs/analysis2/lib/python3.10/site-packages/allel/model/ndarray.py", line 1530, in subset                                                                                          
    return subset_genotype_array(self, sel0, sel1, cls=type(self), subset=subset)                                                                                                                                  
  File "/home/ssmall/mambaforge/envs/analysis2/lib/python3.10/site-packages/allel/model/generic.py", line 227, in subset_genotype_array                                                                            
    out = subset(g.values, sel0, sel1, **kwargs)                                                                                                                                                                   
  File "/home/ssmall/mambaforge/envs/analysis2/lib/python3.10/site-packages/allel/model/ndarray.py", line 69, in subset                                                                                            
    return data[sel0, sel1]                                                                                                                                                                                        
IndexError: arrays used as indices must be of integer (or boolean) type 
```

unless this is an easy fix, maybe not worth recording the correct chrom name in the diploshic output?

Reply:
yeah let's hold off and get this PR merged first.